### PR TITLE
Add kustomization and transformer references

### PR DESCRIPTION
**Pull request by ncapps**
Relates to #5379

### CODE REVIEW

File: site/content/en/docs/Reference/API/Kustomization File/namePrefix.md [hunk 1]
```diff
@@ -4,7 +4,18 @@ linkTitle: "namePrefix"
...
 
-The Tasks section contains examples of how to use [`namePrefix`](/docs/tasks/namespaces_and_names/).
+See the [Tasks section] for examples of how to use `namePrefix`.
+
+### namePrefix
+Add prefixes to the names of all resources.
+
+* **namePrefix** (string)
+
+    NamePrefix will add a prefix the names of all resources mentioned in the Kustomization file including generated resources such as ConfigMaps and Secrets.
```
Comment (antoooks) [inline site/content/en/docs/Reference/API/Kustomization File/namePrefix.md:18]:
```suggestion
    NamePrefix will add a prefix the names of all resources mentioned in the Kustomization file including generated resources such as ConfigMaps and Secrets.
```
nit.
prefix `to` the names

File: site/content/en/docs/Reference/API/Kustomization File/nameSuffix.md [hunk 2]
```diff
@@ -4,7 +4,18 @@ linkTitle: "nameSuffix"
...
 
-The Tasks section contains examples of how to use [`nameSuffix`](/docs/tasks/namespaces_and_names/).
+See the [Tasks section] for examples of how to use `nameSuffix`.
+
+### nameSuffix
+Add suffixes to the names of all resources.
+
+* **nameSuffix** (string)
+
+    NameSuffix will add a suffix the names of all resources mentioned in the Kustomization file including generated resources such as ConfigMaps and Secrets.
```
Comment (antoooks) [inline site/content/en/docs/Reference/API/Kustomization File/nameSuffix.md:18]:
```suggestion
    NameSuffix will add a suffix the names of all resources mentioned in the Kustomization file including generated resources such as ConfigMaps and Secrets.
```

nit. suffix `to` the names

File: site/content/en/docs/Reference/API/Kustomization File/namePrefix.md (hunk 1 above)
Comment (ncapps) [reply site/content/en/docs/Reference/API/Kustomization File/namePrefix.md]: Thanks for your review! I updated this per your suggestion. Resolving this comment.

File: site/content/en/docs/Reference/API/Kustomization File/nameSuffix.md (hunk 2 above)
Comment (ncapps) [reply site/content/en/docs/Reference/API/Kustomization File/nameSuffix.md]: Updated per your suggestion. Resolving comment.

File: site/content/en/docs/Reference/API/Kustomization File/labels.md [hunk 3]
```diff
@@ -6,5 +6,35 @@ weight: 10
...
+
+    * **includeSelectors** (bool), optional
+
+        IncludeSelectors indicates whether the transformer should include the fieldSpecs for selectors. Custom fieldSpec specified by `fields` will be merged with builtin fieldSpecs if this is true. Defaults to false.
+
+    * **includeTemplates** (bool), optional
+
+        IncludeTemplates indicates whether the transformer should include the `spec/template/metadata` fieldSpec. Custom fieldSpecs specified by `fields` will be merged with the `spec/template/metadata` fieldSpec if this is true. If IncludeSelectors is true, IncludeTemplates is not needed. Defaults to false.
+
+    * **fieldSpecs** (\[\][FieldSpec]({{< relref "../Common%20Definitions/FieldSpec.md" >}})), optional
```
Comment (stormqueen1990) [inline site/content/en/docs/Reference/API/Kustomization File/labels.md:34]: Is the goal here to list the field how it's called in code, or how a user would use it?

If it's the latter, this should be `fields` instead:
```suggestion
    * **fields** (\[\][FieldSpec]({{< relref "../Common%20Definitions/FieldSpec.md" >}})), optional
```
as specified in the Go field tags: https://github.com/kubernetes-sigs/kustomize/blob/3e69c2e36a5b9d7959c66f7ed39d54dec67dd8ae/api/types/labels.go#L19

Comment (ncapps) [reply site/content/en/docs/Reference/API/Kustomization File/labels.md:34]: I believe the goal is to represent how a user can use this field in a kustomzation file. I updated this to `fields`. Thanks for catching this!

**Review by stormqueen1990 — COMMENTED** [review]
/lgtm

Thanks @ncapps, this addresses issue #5549 🙏

**Pull request merged**